File: blind75/ContainsDuplicate.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    /*
    Returns true if there are no duplicates of any element in the vector.
    False if there exists a duplicate/copy.
    */
    bool containsDuplicate(vector<int>& nums) {
        // unordered map to hold occurences of the elements in nums
	unordered_map<int, int> umap;
        
	// iterate through the vector and increment the value(occurrence) of the key(which is the number in the vector)
        for(auto x : nums){
            umap[x]++;
            if(umap[x] > 1){
                return true;
            }
        }
        return false;
    }
};

int main() {
	Solution s;
	vector<int> nums = {1,2,3,4};
	cout << s.containsDuplicate(nums);
	return 0;
}

File: blind75/contains_duplicate.py
from typing import List

def containsDuplicate(nums:List[int]) -> bool:
    
    # to keep elements that occured once
    checked = set()

    for number in nums:
        # number occured already in the set that means its a duplicate
        if number in checked:
            return True
        checked.add(number)

    return False

def main():
    nums = [1,2,3,4,5,1]
    print(containsDuplicate(nums))

if __name__ == "__main__":
    main()